src/Bicep.Core/Diagnostics/DiagnosticBuilder.cs

```diff
@@ -943,6 +943,28 @@ public ErrorDiagnostic ArgumentCountMismatch(int argumentCount, int mininumArgum
                 TextSpan,
                 "BCP166",
                 $"Duplicate \"{decoratorName}\" decorator.");
+
+            public ErrorDiagnostic ResourceTypeIsNotValidParent(string resourceType, string parentResourceType) => new(
+                TextSpan,
+                "BCP167",
+                $"Resource type \"{resourceType}\" is not a valid child resource of parent \"{parentResourceType}\".");
```

Comment:
>parent [](start = 84, length = 6)

Should the message also say what the permissible types would be? (I haven't yet read how this is used.)

Reply:
Personally I don't think we should list the available child types in the error - the inconsistency is detected via prefix matching (using the type string to check if it's a child), so it feels like it could be inconsistent or potentially misleading to try and detect possible types from the type system.